src/components/FilterModal/FilterModal.jsx

```diff
@@ -1,25 +1,51 @@
-import React, { useState } from 'react';
+import React, { useState, useRef } from 'react';
 import PropTypes from 'prop-types';
 import { Button } from 'react-bootstrap';
+import { useDispatch } from 'react-redux';
 import Modal from '../Modal/Modal';
 import CheckBoxList from '../CheckBoxList/CheckBoxList';
+import { getFilterSetter, getFilterResetter } from '../../filters/freaksFilter';
 
-function FilterModal({ title, isOpen, onClose, keywords }) {
+function FilterModal({ title, isOpen, onClose, keywords, setOpenModal }) {
   const [ checkedState, setCheckedState ] = useState([]);
+  const queuedFilters = useRef([]);
+  const dispatch = useDispatch();
+
+  const onChangeCb = (result) => {
+    setCheckedState(result);
+    queuedFilters.current = [ ...result ];
+  };
+
+  const onClickCb = () => {
```

Comment:
No need to suffix callback with `Cb` as we already know that they are callbacks. Also in the jsx we understand that the function handles the click, but exactly not what it does. These kind of functions should somehow reflect that.

This for example applies the filters, so should be something like `applyFilters`